src/main/java/roomescape/service/TimeService.java

```diff
@@ -0,0 +1,49 @@
+package roomescape.service;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import org.springframework.stereotype.Service;
+import roomescape.domain.Time;
+import roomescape.dto.TimeResponseDto;
+import roomescape.dto.TimeSaveRequestDto;
+import roomescape.repository.TimeDAO;
+
+@Service
+public class TimeService {
+
+    private final TimeDAO timeDAO;
+
+    public TimeService(TimeDAO timeDAO) {
+        this.timeDAO = timeDAO;
+    }
+
+    public TimeResponseDto createTime(TimeSaveRequestDto requestDto) {
+        if (isTimeArgumentEmpty(requestDto)) {
+            throw new IllegalArgumentException("잘못된 요청입니다.");
+        }
+
+        Time time = new Time(requestDto.getTime());
+        Long id = timeDAO.insert(time);
+
+        return new TimeResponseDto(id, time);
+    }
+
+    public List<TimeResponseDto> readAllTimes() {
+        return timeDAO.findAllTimes()
+                .stream()
+                .map(TimeResponseDto::new)
+                .collect(Collectors.toList());
+    }
```

Comment:
```suggestion
    public List<TimeResponseDto> readAllTimes() {
        return timeDAO.findAllTimes()
                .stream()
                .map(TimeResponseDto::new)
                .toList();
    }
```
- `collect(Collectors.toList())`와 `toList()`의 차이를 학습하고 알려주세요!
- 저는 `toList()`가 UnmodifiableList를 반환하기 때문에 자주 사용해요!

Reply:
## 🤓 `collect(Collectors.toList())`와 `toList()`

### ✍️ `collect(Collectors.toList())`, `collect(Collectors.toUnmodifiableList())`, `toList()` 표 비교
|버전|`java8`|`java10`|`java16`|
|---|---|---|---|
|List 변환 방법|`.collect(Collectors.toList())`|`.collect(Collectors.toUnmodifiableList())`|`.toList()`|
|반환 타입|`ArrayList`|`UnmodifiableList`|`UnmodifiableList`|
|수정 가능 여부|O|X|X|
|`Null` 허용 여부|O|X|O|

### `collect(Collectors.toList())` -> `toList()` 변경의 장점
`collect(Collectors.toList())` 는 `ArrayList` 를 반환하고
`toList()` 는 `Collectors.UnmodifiableList` 또는 `Collectors.UnmodifiableRandomAccessList` 를 반환합니다.
두 반환 타입의 차이점은 수정 가능 여부입니다.
`ArrayList`는 수정이 가능하고 `UnmodifiableList`는 수정이 불가능합니다.
`UnmodifiableList`는 수정하려고 하면 `UnsupportedOperationException` 예외가 발생합니다.
즉, `UnmodifiableList`는 읽기 용도로만 사용 가능합니다.

### `Unmodifiable Collection`
- 외부에서 변경 시 예외처리(`UnsupportedOperationException`)되기 때문에 예기치 않은 값 변경 방지합니다.
- getter로 값을 꺼내도 데이터 수정 불가능합니다.
- 다만, `Unmodifiable`과 `Immutable`은 다릅니다. `Unmodifiable`은 **수정을 막는 것**이 핵심 기능이므로 원본 객체와 주소가 공유되어 있고, 원본 객체가 변경되면 동일하게 영향을 받습니다.


---
### 참고 사이트
- https://velog.io/@cieroyou/Stream%EC%9D%84-List%EB%A1%9C-%EB%B3%80%ED%99%98%ED%95%98%EB%8A%94-%EB%8B%A4%EC%96%91%ED%95%9C-%EB%B0%A9%EB%B2%95%EA%B3%BC-%EC%B0%A8%EC%9D%B4Collectors.toList-vs-Stream.toList
- https://binux.tistory.com/146
- https://hello-judy-world.tistory.com/209
- https://tecoble.techcourse.co.kr/post/2021-04-26-defensive-copy-vs-unmodifiable/
---
### 추가로 공부해야 하는 키워드
- 방어적 복사, 깊은 복사, 얕은 복사
- 불변 객체
- copy.of
- https://pparksean.tistory.com/122?category=986364
https://ksh-coding.tistory.com/77

Reply:
`.collect(Collectors.toList())` -> `.toList()`로 코드도 수정했습니다!